nemo/core/classes/common.py

```diff
@@ -41,7 +42,7 @@ def output_types(self) -> Optional[Dict[str, NeuralType]]:
         """Define these to enable output neural type checks"""
         return None
 
-    def _validate_input_types(self, **kwargs):
+    def _validate_input_types(self, input_types, **kwargs):
```

Comment:
why does input_types does not have default (None) here? 
E.g. why not:
_validate_input_types(self, input_types=None, **kwargs):

Reply:
Yep this can be done to. I'll default to using self.input_types if it is None (which can itself be None too)

Reply:
Ok so I can set it default to None, but cant inject self.input_types into this. There can be cases where, (cls.input_types is not None), but (func_override.input_types is None). This is a reasonable default to not inject, at least for the moment.